WalletWasabi.Tests/UnitTests/WabiSabi/Integration/WabiSabiHttpApiIntegrationTests.cs

```diff
@@ -531,13 +531,13 @@ public async Task CoinJoinWithBlameRoundTestAsync(long[] amounts)
 
 					// In case some participants claim to have finished successfully then wait a second for seeing
 					// the coinjoin in the mempool. This seems really hard to believe but just in case.
-					if (participantsFinishedSuccessully.All(x => x.SuccessfulBroadcast))
+					if (participantsFinishedSuccessully.All(x => x is SuccessfulCoinjoinResult))
 					{
 						await Task.Delay(TimeSpan.FromSeconds(1));
 						var mempool = await rpc.GetRawMempoolAsync();
 						Assert.Single(mempool);
 					}
-					else if (participantsFinishedSuccessully.All(x => x.GoForBlameRound == false && x.SuccessfulBroadcast == false))
+					else if (participantsFinishedSuccessully.All(x => x is FailedCoinjoinResult))
```

Comment:
Yes, I reverted the inheritance because it sound natural but I really want something for like Discriminated Unions here where I am "forced" to consider all the cases one by one.